src/mbgl/gl/gl_object_store.cpp

```diff
@@ -61,12 +61,11 @@ void TexturePoolHolder::create(GLObjectStore& objectStore_) {
 
 void TexturePoolHolder::reset() {
     if (!bool()) return;
-    for (GLuint id : ids) {
-        if (id) {
-            objectStore->abandonedTextures.push_back(id);
-        }
-    }
-    ids.fill(0);
+    std::for_each(ids.begin(), ids.end(), [&](GLuint& id) {
```

Comment:
You are right, I can achieve the same thing capturing a non-const reference in a range-based `for`.